### DISCUSSION TITLE
show all the failed tests in the "PROBLEMS" tab

### DISCUSSION THREAD

**Discussion by EliTheSurfer** [vitest-dev/vscode#442] [Q&A]
Hello everyone,
I was wondering if there is any way to show all the failed tests in the "problems" tab ?

![image](https://github.com/user-attachments/assets/164d2586-edfa-4a73-ac11-55ccc1390d3b)

If any one knows how to achieve this, this would be a great help 🙏

**Comment by sheremet-va** [comment]
I think this can be achieved if this is implemented: https://github.com/vitest-dev/vscode/issues/431

**Reply by EliTheSurfer** [reply]
I have implemented a solution using [VSCode's diagnostic solution](https://code.visualstudio.com/api/references/vscode-api#DiagnosticCollection), and it works like a charm. I can even navigate or loop through failed tests using a keyboard shortcut: `editor.action.marker.next` (Go to Next Problem).

If others are interested, I can make a pull request.

**Reply by sheremet-va** [reply]
Sure, PR is welcome

This is where we set results: https://github.com/vitest-dev/vscode/blob/bf57adc0e257ad72033e92bc71db38c36c4d7fed/src/runner/runner.ts#L115
